### ISSUE TITLE
Object destructuring for variable declaration

### ISSUE THREAD

**Issue by jodranata** [eslint/eslint#13223]
**Tell us about your environment**

* **ESLint Version: ^6.8.0**
* **Node Version: v.13.13.0**
* **npm Version: 6.14.4**

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "allowImportExportEverywhere": false,
    "codeFrame": false,
    "ecmaFeatures": {
      "jsx": true,
      "destructuring": true
    },
    "ecmaVersion": 2018
  },
  "plugins": ["prettier"],
  "extends": ["airbnb", "prettier"],
  "env": {
    "browser": true,
    "jest": true
  },
  "rules": {
    "max-len": ["error", { "code": 80 }],
    "prefer-promise-reject-errors": ["off"],
    "react/jsx-filename-extension": ["off"],
    "react/prop-types": ["warn"],
    "react/destructuring-assignment": [2, "never"],
    "no-return-assign": ["warn"],
    "no-unused-vars": "off",
    "no-plusplus": "off",
    "consistent-return": "off",
    "prefer-destructuring": [
      "error",
      {
        "VariableDeclarator": {
          "array": true,
          "object": true
        }
      },
      {
        "enforceForRenamedProperties": true
      }
    ],
    "no-useless-rename": ["error", { "ignoreDestructuring": true }],
    "react/prefer-stateless-function": ["warn"],
    "prettier/prettier": ["error"],
    "jsx-a11y/label-has-associated-control": [
      "error",
      {
        "required": {
          "some": ["nesting", "id"]
        }
      }
    ]
  },
  "settings": {
    "react": {
      "pragma": "React",
      "version": "detect"
    }
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const handleChange = (e) => {
    e.preventDefault();
    if (e.target.type === 'number') setNominal(e.target.value);
    else setDetails(e.target.value);
  };
```

```bash

```

**What did you expect to happen?**
Giving error to destructure type & value from e.target before using.

**What actually happened? Please include the actual, raw output from ESLint.**
No errors

**Are you willing to submit a pull request to fix this bug?**
